File: backend/app/models/document.py
from datetime import datetime
from . import db

class Document(db.Model):
    __tablename__ = "documents"

    id = db.Column(db.String(64), primary_key=True)
    bid_id = db.Column(db.String(64), db.ForeignKey("bids.id"), nullable=False)
    document_type = db.Column(db.String(100), nullable=False, default="OTHER")
    original_filename = db.Column(db.String(255), nullable=False)
    storage_key = db.Column(db.String(512), nullable=False)
    mime_type = db.Column(db.String(100), nullable=False, default="application/pdf")
    size_bytes = db.Column(db.Integer, nullable=False, default=0)
    sha256 = db.Column(db.String(64), nullable=False, index=True)
    page_count = db.Column(db.Integer, nullable=True)
    uploaded_by = db.Column(db.String(64), nullable=True)
    processing_status = db.Column(db.String(50), nullable=False, default="UPLOADED")
    description = db.Column(db.String(512), nullable=True)
    created_at = db.Column(db.DateTime, default=datetime.utcnow)
    updated_at = db.Column(db.DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)

    # Relationships
    bid = db.relationship("Bid", back_populates="documents")

    # Backwards compatibility properties
    @property
    def filename(self) -> str:
        return self.original_filename

    @property
    def category(self) -> str:
        return self.document_type

    @property
    def file_path(self) -> str:
        return self.storage_key

    @property
    def file_size_bytes(self) -> int:
        return self.size_bytes

    def to_dict(self):
        return {
            "id": self.id,
            "bid_id": self.bid_id,
            "document_type": self.document_type,
            "original_filename": self.original_filename,
            "mime_type": self.mime_type,
            "size_bytes": self.size_bytes,
            "sha256": self.sha256,
            "page_count": self.page_count,
            "uploaded_by": self.uploaded_by,
            "processing_status": self.processing_status,
            "description": self.description,
            "created_at": self.created_at.isoformat() if self.created_at else None,
            "updated_at": self.updated_at.isoformat() if self.updated_at else None
        }

File: frontend-old/src/types/document.ts
export type DocumentType =
  | 'GST_CERTIFICATE'
  | 'PAN_CARD'
  | 'UDYAM_CERTIFICATE'
  | 'OEM_AUTHORIZATION'
  | 'FINANCIAL_STATEMENT'
  | 'TURNOVER_CERTIFICATE'
  | 'TECHNICAL_SPECIFICATION'
  | 'MAKE_IN_INDIA_DECLARATION'
  | 'EXPERIENCE_CERTIFICATE'
  | 'OTHER';

export const DOCUMENT_TYPE_LABELS: Record<DocumentType, string> = {
  GST_CERTIFICATE: 'GST Certificate',
  PAN_CARD: 'PAN Card',
  UDYAM_CERTIFICATE: 'Udyam Certificate',
  OEM_AUTHORIZATION: 'OEM Authorization',
  FINANCIAL_STATEMENT: 'Financial Statement',
  TURNOVER_CERTIFICATE: 'Turnover Certificate',
  TECHNICAL_SPECIFICATION: 'Technical Specification',
  MAKE_IN_INDIA_DECLARATION: 'Make in India Declaration',
  EXPERIENCE_CERTIFICATE: 'Experience Certificate',
  OTHER: 'Other Document',
};

export const EXPECTED_BID_DOCUMENTS: DocumentType[] = [
  'GST_CERTIFICATE',
  'PAN_CARD',
  'UDYAM_CERTIFICATE',
  'OEM_AUTHORIZATION',
  'FINANCIAL_STATEMENT',
  'TECHNICAL_SPECIFICATION',
];

export interface Document {
  id: string;
  bid_id?: string;
  tender_id?: string;
  document_type: DocumentType | string;
  original_filename: string;
  mime_type: string;
  size_bytes: number;
  sha256: string;
  page_count?: number | null;
  description?: string | null;
  processing_status: 'UPLOADED' | 'PROCESSING' | 'PROCESSED' | 'FAILED' | 'INVALID' | string;
  uploaded_by?: string;
  created_at?: string;
  updated_at?: string;
  // Legacy optional properties for backward compatibility
  name?: string;
  document_id?: string;
  url?: string;
  storage_key?: string;
  uploadedAt?: string;
  processing_error?: string | null;
  ocr_engine?: string | null;
  pages?: OCRPage[];
  classified_document_type?: string | null;
  classification_confidence?: number | null;
  classification_status?: 'CLASSIFIED' | 'UNKNOWN' | 'PROCESSING' | 'FAILED' | 'REVIEW_REQUIRED' | string | null;
  classification_method?: string | null;
  classified_at?: string | null;
  classification_error?: string | null;
}

export interface OCRPage {
  page_number: number;
  text: string;
  ocr_confidence?: number | null;
}
